src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/AddAzureRmSqlServerTransparentDataEncryptionCertificate.cs

```diff
@@ -0,0 +1,211 @@
+            Position = 0,
+            HelpMessage = "The sql server input object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlServerModel SqlServer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Server Resource Id
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The sql server resource id")]
+        [ValidateNotNullOrEmpty]
+        public string SqlServerResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The Resource Group Name")]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent server name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 1,
+            HelpMessage = "The Server Name")]
+        [ValidateNotNullOrEmpty]
+        public virtual string ServerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the PrivateBlob for Transparent Data Encryption Certificate
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            Position = 1,
+            HelpMessage = PrivateBlobHelpText)]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            Position = 1,
+            HelpMessage = PrivateBlobHelpText)]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 2,
+            HelpMessage = PrivateBlobHelpText)]
+        [ValidateNotNullOrEmpty]
+        public string PrivateBlob { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Password for Transparent Data Encryption Certificate
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "The Password for Transparent Data Encryption Certificate")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "The Password for Transparent Data Encryption Certificate")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "The Password for Transparent Data Encryption Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string Password { get; set; }
```

Comment:
@nivimsft same comment about making this a `SecureString`

Reply:
Addressed